tests/unit/amo/pages/TestAddon.js

```diff
@@ -620,6 +620,30 @@ describe(__filename, () => {
     sinon.assert.callCount(fakeDispatch, 1);
   });
 
+  // See: https://github.com/mozilla/addons-frontend/issues/4271.
+  it("dispatches a server redirect when slug param case does not match the add-on's slug case", () => {
+    const clientApp = CLIENT_APP_FIREFOX;
+    const { store } = dispatchClientMetadata({ clientApp });
+    const addon = createInternalAddon(fakeAddon);
+    store.dispatch(_loadAddons({ addon }));
+
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+    renderComponent({
+      // Set the slug param to all uppercase
+      params: { slug: addon.slug.toUpperCase() },
```

Comment:
Nit: There's no guarantee that `slug` in `fakeAddon` isn't already all upper case, so we should probably set a manual value for `slug` in the test. Of course, the test would fail if that were the case, but it still feels like the right thing to do, to me.

Reply:
well, calling the method implies that it is not already the case, but I can of course be more verbose and create an addon with a lowercase slug.